app/server.go

```diff
@@ -1454,6 +1465,93 @@ func doCheckWarnMetricStatus(a *App) {
 	}
 }
 
+func (s *Server) StopMetricsServer() {
+	s.metricsLock.Lock()
+	defer s.metricsLock.Unlock()
+
+	if s.metricsServer != nil {
+		s.metricsServer.Close()
+		s.Log.Info("Metrics and profiling server is stopping", mlog.String("address", *s.Config().MetricsSettings.ListenAddress))
+		s.metricsServer = nil
+	}
+}
+
+func (s *Server) HandleMetrics(route string, h http.Handler) {
+	if s.metricsRouter != nil {
+		s.metricsRouter.Handle(route, h)
+	}
+}
+
+func (s *Server) InitMetricsRouter() {
+	if s.metricsRouter != nil {
+		return
+	}
+
+	s.metricsRouter = mux.NewRouter()
+	runtime.SetBlockProfileRate(*s.Config().MetricsSettings.BlockProfileRate)
+
+	rootHandler := func(w http.ResponseWriter, r *http.Request) {
+		html := `
+			<html>
+				<body>
+					<div><a href="/metrics">Metrics</a></div>
+					<div><a href="/debug/pprof/">Profiling Root</a></div>
+					<div><a href="/debug/pprof/cmdline">Profiling Command Line</a></div>
```

Comment:
While we are here, let's add the /profile and /trace endpoints in the HTML response.